cmd/snap/cmd_info.go

```diff
@@ -321,8 +322,19 @@ func maybePrintServices(w io.Writer, snapName string, allApps []client.AppInfo,
 var channelRisks = []string{"stable", "candidate", "beta", "edge"}
 
 // displayChannels displays channels and tracks in the right order
-func displayChannels(w io.Writer, chantpl string, esc *escapes, remote *client.Snap) {
-	fmt.Fprintf(w, "channels:"+strings.Repeat("\t", strings.Count(chantpl, "\t"))+"\n")
+func (x *infoCmd) displayChannels(w io.Writer, chantpl string, esc *escapes, remote *client.Snap, revLen, sizeLen int) (maxRevLen, maxSizeLen int) {
```

Comment:
I wonder how much overkill would be to have a struct that carries the chantpl,  the array of chInfoTs and the max lens with a couple of methods to add infos or print them or consider sizes, vs passing the sizes in and out?

Reply:
something like that sounds good. I'd rather wait to do that until after epochs (or next week), though.